cache/cache.go

```diff
@@ -1,65 +1,62 @@
 package cache
 
 import (
+	"io"
 	"os"
 	"path/filepath"
+	"strings"
 
-	bolt "github.com/simar7/gokv/bbolt"
-	"github.com/simar7/gokv/encoding"
-	kvtypes "github.com/simar7/gokv/types"
 	"golang.org/x/xerrors"
 )
 
+const (
+	cacheDirName = "fanal"
+)
+
+var (
+	replacer = strings.NewReplacer("/", "_")
+)
+
 type Cache interface {
-	Get(bucket, key string, value *[]byte) (found bool, err error)
-	Set(bucket, key string, value []byte) (err error)
+	Get(key string) io.Reader
+	Set(key string, file io.Reader) (io.Reader, error)
 	Clear() error
 }
 
-type RealCache struct {
-	directory string
-	cache     *bolt.Store
+type FSCache struct {
+	Directory string
```

Comment:
Do we need to export this value? If we always pass this value via `Initialize`, we should unexport it.

```suggestion
	directory string
```